proto/meta.proto

```diff
@@ -544,9 +566,38 @@ message TableParallelism {
   }
 }
 
+// Changes a streaming job in place by
+// overwriting the labels of a materialized view/table/sink with the given id
+message UpdateStreamingJobRequest {
+  // Id of the materialized view, table, or sink which we want to update
+  uint32 id = 1;
+
+  // replace the node_labels map of the streaming job with a given id 
+  // with below map
+  map<string, string> node_labels = 2;
+}
+
+// We do not need to add an explicit status field here, we can just use the RPC status
+message UpdateStreamingJobResponse {}
+
+message GetStreamingJobsStatusRequest {}
+
+// Descriptions of MVs and sinks
+message GetStreamingJobsStatusResponse {
+  message Status {
+    uint32 table_id = 1;
+    map<string, string> node_labels = 2;
+    bool backfill_done = 3;
+  }
+
+  repeated Status streaming_job_statuses = 1;
+}
+
 service ScaleService {
   rpc GetClusterInfo(GetClusterInfoRequest) returns (GetClusterInfoResponse);
   rpc Reschedule(RescheduleRequest) returns (RescheduleResponse);
+  rpc UpdateStreamingJob(UpdateStreamingJobRequest) returns (UpdateStreamingJobResponse); 
```

Comment:
The naming of `UpdateStreamingJob` is too general.

Reply:
Changing it to 
```
rpc UpdateStreamingJobNodeLabels(UpdateStreamingJobNodeLabelsRequest) returns (UpdateStreamingJobNodeLabelsResponse); 
```